File: scripts/with_test_translation_outputs/Translation/Avatar/codellama/CodeLlama-34b-Instruct-hf/Python/ed896bf7451f8d4b169710495be074ec3feed85d/Experiment_Results/intermediate/SR/Translation/use_test/CodeLlama-34b-Instruct-hf/Avatar/Python/Java/atcoder_ABC132_F.java
<s> [INST] <<SYS>>
You are an expert Java programmer and assistant
<</SYS>>

Translate the following Python code to Java and enclose your solution inside ```java and ```.
A sample test case is provided below:

Test input:
1000000000 100

Expected output:
926307829


```
mod = 10 ** 9 + 7
n , k = map ( int , input ( ).split ( ) )
coef = [ ]
for i in range ( 1 , int ( n ** 0.5 ) + 1 ) :
    coef.append ( ( n // i ) - ( n // ( i + 1 ) ) )
coef += [ 1 ] * ( n - sum ( coef ) )
coef = coef [ : : - 1 ]
nn = len ( coef )
dp = [ [ 0 ] * nn for _ in range ( k + 1 ) ]
dp [ 0 ] [ 0 ] = 1
for i in range ( 1 , k + 1 ) :
    tmp = 0
    for j in range ( nn - 1 , - 1 , - 1 ) :
        tmp += dp [ i - 1 ] [ ( nn - 1 ) - j ]
        tmp %= mod
        dp [ i ] [ j ] = coef [ j ] * tmp
        dp [ i ] [ j ] %= mod
print ( sum ( dp [ k ] ) % mod )


```
[/INST]
```java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        int mod = 1000000007;

        int[] coef = new int[n];
        for (int i = 1; i <= n; i++) {
            coef[i - 1] = (n / i) - (n / (i + 1));
        }
        coef[n - 1] = 1;

        int[][] dp = new int[k + 1][n];
        dp[0][0] = 1;
        for (int i = 1; i <= k; i++) {
            int tmp = 0;
            for (int j = n - 1; j >= 0; j--) {
                tmp += dp[i - 1][(n - 1) - j];
                tmp %= mod;
                dp[i][j] = coef[j] * tmp;
                dp[i][j] %= mod;
            }
        }

        int result = 0;
        for (int i = 0; i < n; i++) {
            result += dp[k][i];
            result %= mod;
        }

        System.out.println(result);
    }
}
```
Note: The above code uses the modular arithmetic property to avoid overflow. The modulus is set to 1000000007, which is a prime number.</s>

File: dataset/Intermediate/Translation/Avatar/v1-Python-code/atcoder_ABC132_F.py
import datetime
import numpy as np
mod = 10 ** 9 + 7
(n, k) = map(int, input().split())
coef = []
for i in range(1, int(n ** 0.5) + 1):
    coef.append(n // i - n // (i + 1))
coef += [1] * (n - np.sum(np.array([coef])))
coef = coef[::-1]
nn = len(coef)
datetime.datetime.now()
dp = [[0] * nn for _ in range(k + 1)]
dp[0][0] = 1
for i in range(1, k + 1):
    tmp = 0
    for reversed_index in range(nn - 1, -1, -1):
        tmp += dp[i - 1][nn - 1 - reversed_index]
        tmp = tmp % mod
        dp[i][reversed_index] = coef[reversed_index] * tmp
        dp[i][reversed_index] %= mod
print(sum(dp[k]) % mod)